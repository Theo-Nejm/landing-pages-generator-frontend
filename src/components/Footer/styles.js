import styled, { css } from 'styled-components';

export const Container = styled.footer`
  ${({ theme }) => css`
    p,
    a,
    i,
    b {
      font-size: ${theme.fonts.sizes.xxsmall};
      text-align: center;
    }
  `}
`;
