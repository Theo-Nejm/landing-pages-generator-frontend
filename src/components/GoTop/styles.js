import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    color: #fff;
    background: ${theme.colors.primaryColor};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    border-radius: 50%;
    position: fixed;
    bottom: 2rem;
    right: 2rem;
    z-index: 4;
    opacity: 0.85;
  `}
`;
