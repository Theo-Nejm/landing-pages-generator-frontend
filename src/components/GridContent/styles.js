import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css`
    text-align: center;
    max-width: 58rem;
  `}
`;

export const HTMLContainer = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.xlarge} 0;
  `}
`;
