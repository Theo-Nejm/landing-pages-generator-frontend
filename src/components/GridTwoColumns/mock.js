export default {
  title: 'Título da seção',
  text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Excepturi iusto fugiat quos nobis nulla enim placeat eius nam quasi sint autem ad corporis, odit, cumque minus esse ipsam, quidem voluptate.',
  srcImg:
    'https://t.ctcdn.com.br/8FplhVkDQdAatiUcehCimgkGJlI=/512x288/smart/i257652.jpeg',
};
