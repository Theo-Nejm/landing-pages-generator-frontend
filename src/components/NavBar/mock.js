export default [
  {
    children: 'Link 1',
    link: '#target-1',
    newTab: false,
  },
  {
    children: 'Link 2',
    link: '#target-2',
    newTab: false,
  },
  {
    children: 'Link 3',
    link: '#target-3',
    newTab: false,
  },
  {
    children: 'Link 4',
    link: '#target-4',
    newTab: false,
  },
  {
    children: 'Link 5',
    link: '#target-5',
    newTab: false,
  },
];
