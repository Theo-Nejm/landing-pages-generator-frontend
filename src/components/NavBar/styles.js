import styled, { css } from 'styled-components';

export const Container = styled.nav`
  ${({ theme }) => css`
    display: flex;
    flex-flow: row wrap;
    background: #fff;

    @media ${theme.media.medium} {
      flex-flow: column nowrap;
      align-items: center;
    }
  `}
`;
