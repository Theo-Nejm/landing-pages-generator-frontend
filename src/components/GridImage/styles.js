import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    h2 {
      font-size: 4.2rem;
    }

    ${TextComponent} {
      margin-bottom: ${theme.spacings.huge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
  `}
`;

export const Element = styled.div`
  ${() => css`
    overflow: hidden;

    > img {
      width: 100%;
      transition: all 0.3s ease-in-out;

      &:hover {
        transform: scale(1.2) rotate(10deg);
      }
    }
  `}
`;
