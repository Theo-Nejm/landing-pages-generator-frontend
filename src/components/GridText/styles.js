import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column nowrap;
    margin-top: 6rem;

    h2 {
      font-size: 4.2rem;
    }

    ${TextComponent} {
      margin-bottom: ${theme.spacings.huge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};

    @media ${theme.media.medium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const Element = styled.div`
  ${({ theme }) => css`
    ${Heading} {
      position: relative;
      left: 5rem;
    }

    ${Heading}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.fonts.sizes.xhuge};
      left: -5rem;
      top: -2rem;
      transform: rotate(7.5deg);
    }
  `}
`;
