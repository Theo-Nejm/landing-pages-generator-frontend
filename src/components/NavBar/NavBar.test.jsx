import { screen } from '@testing-library/dom';
import { NavBar } from '.';
import { renderTheme } from '../../styles/renderTheme';
import { theme } from '../../styles/theme';

import mock from './mock';

describe('<NavBar />', () => {
  it('should render links', () => {
    renderTheme(<NavBar links={mock} />);
    expect(screen.getAllByRole('link')).toHaveLength(mock.length);
  });

  it("should'nt render links", () => {
    renderTheme(<NavBar />);
    expect(screen.queryAllByText('links')).toHaveLength(0);
  });

  it('should break the layout to responsivity when mobile', () => {
    renderTheme(<NavBar links={mock} />);
    expect(
      screen.getByRole('link', { name: /Link 5/i }).parentElement,
    ).toHaveStyleRule('flex-flow', 'column nowrap', {
      media: theme.media.medium,
    });
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<NavBar links={mock} />);
    expect(container).toMatchSnapshot();
  });
});
