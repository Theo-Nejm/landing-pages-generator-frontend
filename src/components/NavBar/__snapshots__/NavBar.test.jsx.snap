// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.c1 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0a1128;
  position: relative;
}

.c1::after {
  content: '';
  position: absolute;
  bottom: 0.75rem;
  left: 25%;
  height: 1px;
  width: 50%;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  height: 0.2rem;
  background: #dc143c;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
}

.c1:hover::after {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

@media screen and (max-width:768px) {
  .c0 {
    -webkit-flex-flow: column nowrap;
    -ms-flex-flow: column nowrap;
    flex-flow: column nowrap;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<div>
  <nav
    aria-label="Main menu"
    class="c0"
  >
    <a
      class="c1"
      href="#target-1"
      target="_self"
    >
      Link 1
    </a>
    <a
      class="c1"
      href="#target-2"
      target="_self"
    >
      Link 2
    </a>
    <a
      class="c1"
      href="#target-3"
      target="_self"
    >
      Link 3
    </a>
    <a
      class="c1"
      href="#target-4"
      target="_self"
    >
      Link 4
    </a>
    <a
      class="c1"
      href="#target-5"
      target="_self"
    >
      Link 5
    </a>
  </nav>
</div>
`;
